.circleci/continue_config.yml

```diff
@@ -0,0 +1,235 @@
+  all:
+    type: boolean
+    default: false
+  example:
+    type: boolean
+    default: false
+
+jobs:
+  ci-base:
+    parameters:
+      python-version:
+        type: string
+        default: "3.9"
+    docker:
+      - image: cimg/python:<< parameters.python-version >>
+    resource_class: small
+    environment:
+      POETRY_CACHE_DIR: /home/circleci/project/.poetry
+    steps:
+      - checkout
+      - restore_cache:
+          key: &ci-base-cache ci-cache-<< parameters.python-version >>-{{ checksum "pyproject.toml" }}
+      - run: |
+          poetry config experimental.new-installer false
+          poetry config installer.max-workers 10
+          poetry install --no-ansi
+      - save_cache:
+          key: *ci-base-cache
+          paths:
+            - /home/circleci/project/.poetry/virtualenvs
+            - poetry.lock
+
+  unit-test:
+    parameters:
+      python-version:
+        type: string
+        default: "3.9"
+    docker:
+      - image: cimg/python:<< parameters.python-version >>
+    resource_class: small
+    environment:
+      POETRY_CACHE_DIR: /home/circleci/project/.poetry
+    steps:
+      - checkout
+      - restore_cache:
+          key: ci-cache-<< parameters.python-version >>-{{ checksum "pyproject.toml" }}
+      - run: poetry run python3 -c 'import kolena'
+      # TODO: fix underlying mypy issues with Python>3.9 rather than skipping
+      - when:
+          condition:
+            not:
+              or:
+                - equal: [ "3.10", << parameters.python-version >> ]
+                - equal: [ "3.11", << parameters.python-version >> ]
+          steps:
+            - run: poetry run pre-commit run -a
+      - run:
+          name: Run unit tests
+          command: |
+            poetry run pytest -vv --cov=kolena --cov-branch tests/unit
+      - when:
+          # Generate coverage only from one python version
+          condition:
+            equal: [ "3.9", << parameters.python-version >> ]
+          steps:
+            - run:
+                name: Coverage
+                command: |
+                  poetry run coverage xml --data-file .coverage
+            - codecov/upload:
+                file: coverage.xml
+                flags: integration
+
+  integration-test:
+    parameters:
+      python-version:
+        type: string
+        default: "3.9"
+      pytest-group:
+        type: string
+        default: "generic"
+      enabled:
+        type: boolean
+        default: false
+      parallelism:
+        type: integer
+        default: 1
+    docker:
+      - image: cimg/python:<< parameters.python-version >>
+    resource_class: small
+    environment:
+      POETRY_CACHE_DIR: /home/circleci/project/.poetry
+    parallelism: << parameters.parallelism >>
+    steps:
+      - when:
+          condition:
+            not: << parameters.enabled >>
+          steps:
+            - run:
+                name: skip job
+                command: circleci-agent step halt
```

Comment:
Have we tested that the job is properly skipped when not scoped? If so, I'm willing to approve based on the results alone.

Reply:
i have tested several combinations.